server/base/src/main/java/org/apache/accumulo/server/master/balancer/HostRegexTableLoadBalancer.java

```diff
@@ -0,0 +1,314 @@
+      Map<KeyExtent,TServerInstance> assignments) {
+
+    if ((System.currentTimeMillis() - this.lastOOBCheck) > this.oobCheckMillis) {
+      // Check to see if a tablet is assigned outside the bounds of the pool. If so, migrate it.
+      for (Entry<TServerInstance,TabletServerStatus> e : current.entrySet()) {
+        String assignedPool = getPoolNameForHost(e.getKey().host());
+        for (String pool : poolNameToRegexPattern.keySet()) {
+          if (assignedPool.equals(pool) || pool.equals(DEFAULT_POOL)) {
+            continue;
+          }
+          String tid = getTableOperations().tableIdMap().get(pool);
+          try {
+            List<TabletStats> outOfBoundsTablets = getOnlineTabletsForTable(e.getKey(), tid);
+            for (TabletStats ts : outOfBoundsTablets) {
+              LOG.info("Tablet {} is currently outside the bounds of the regex, reassigning", ts.toString());
+              unassigned.put(new KeyExtent(ts.getExtent()), e.getKey());
+            }
+          } catch (TException e1) {
+            LOG.error("Error in OOB check getting tablets for table {} from server {}", tid, e.getKey().host(), e);
+          }
+        }
+      }
+      this.oobCheckMillis = System.currentTimeMillis();
+    }
+
+    Map<String,SortedMap<TServerInstance,TabletServerStatus>> pools = splitCurrentByRegex(current);
+    // separate the unassigned into tables
+    Map<String,Map<KeyExtent,TServerInstance>> groupedUnassigned = new HashMap<String,Map<KeyExtent,TServerInstance>>();
+    for (Entry<KeyExtent,TServerInstance> e : unassigned.entrySet()) {
+      Map<KeyExtent,TServerInstance> tableUnassigned = groupedUnassigned.get(e.getKey().getTableId());
+      if (tableUnassigned == null) {
+        tableUnassigned = new HashMap<KeyExtent,TServerInstance>();
+        groupedUnassigned.put(e.getKey().getTableId(), tableUnassigned);
+      }
+      tableUnassigned.put(e.getKey(), e.getValue());
+    }
+    // Send a view of the current servers to the tables tablet balancer
+    for (Entry<String,Map<KeyExtent,TServerInstance>> e : groupedUnassigned.entrySet()) {
+      Map<KeyExtent,TServerInstance> newAssignments = new HashMap<KeyExtent,TServerInstance>();
+      String tableName = tableIdToTableName.get(e.getKey());
+      String poolName = getPoolNameForTable(tableName);
+      SortedMap<TServerInstance,TabletServerStatus> currentView = pools.get(poolName);
+      getBalancerForTable(e.getKey()).getAssignments(currentView, e.getValue(), newAssignments);
+      assignments.putAll(newAssignments);
+    }
+  }
+
+  @Override
+  public long balance(SortedMap<TServerInstance,TabletServerStatus> current, Set<KeyExtent> migrations, List<TabletMigration> migrationsOut) {
+    long minBalanceTime = 5 * 1000;
+    // Iterate over the tables and balance each of them
+    TableOperations t = getTableOperations();
+    if (t == null)
```

Comment:
You call getTableOperations() in other places in this code, and assume the response won't be null. Why is it treated differently here? Either this check is superfluous, or the other places need to be fixed.